pkg/apis/numaflow/v1alpha1/vertex_types.go

```diff
@@ -228,7 +226,18 @@ func (v Vertex) GetPodSpec(req GetVertexPodSpecReq) (*corev1.PodSpec, error) {
 	envVars := []corev1.EnvVar{
 		{Name: EnvVertexObject, Value: encodedVertexSpec},
 	}
-	envVars = append(envVars, v.commonEnvs()...)
+
+	commonEnvVars := v.commonEnvs()
+	for _, vtx := range req.PipelineSpec.Vertices {
+		if vtx.IsASource() && vtx.Source.Serving != nil {
+			commonEnvVars = append(commonEnvVars,
+				corev1.EnvVar{Name: EnvCallbackEnabled, Value: "true"},
+				corev1.EnvVar{Name: EnvCallbackURL, Value: KeyMetaCallbackURL},
```

Comment:
Is this correct? The value is a constant.

Reply:
We were using this env variable to let the user specify an alternate service URL to which the callback can be sent, if callback sent to the Serving source pod fails. The value I used is incorrect. Also, we do not use it now (after the recent changes in Serving). I will remove the `EnvCallbackURL` env variable.